### PR TITLE
Add support for Remark, Rehype and retext

### LINKED ISSUE #2036: [Icon Request] Support for Remark
- [x] I'm sure this issue is not a _duplicate_.
- [x] I'm submitting an icon request:
  - Type: `filename`
  - Icon Name: `remark`
  - Sample original Icon: logo
  - Filenames: `.remarkrc.js`

- More info: Remark is a tool that works with Markdown. It includes a [parser](https://github.com/remarkjs/remark/tree/master/packages/remark-parse), [linter](https://github.com/remarkjs/remark-lint), and [CLI](https://github.com/remarkjs/remark/tree/master/packages/remark-cli), and itegrates with other tools. The CLI uses a file named _.remarkrc.js_ for its configuration.

### PR DESCRIPTION
**Pull request by KingDarBoja**
_**Fixes #2036**_

**Changes proposed:**

- [X] Add
- [ ] Delete
- [ ] Fix
- [ ] Prepare

Please check the issue for more info about this PR.

### CODE REVIEW

File: src/iconsManifest/supportedExtensions.ts [hunk 1]
```diff
@@ -2748,13 +2748,40 @@ export const extensions: IFileCollection = {
       languages: [languages.red],
       format: FileFormat.svg,
     },
+    { icon: 'registry', extensions: ['reg'], format: FileFormat.svg },
```
Comment (JimiC) [inline src/iconsManifest/supportedExtensions.ts]: Why is this here?

Comment (JimiC) [reply src/iconsManifest/supportedExtensions.ts]: Ah OK. Reordering.

**Comment by wooorm** [comment]
Here are the “official” initials files that I generated, if anyone needs them: 
[initials.zip](https://github.com/vscode-icons/vscode-icons/files/3156341/initials.zip)

I’d also like to add that `.js` and `.yml` / `.yaml` are also valid extensions for the RC files, but the default is JSON (no extension, or `.json`)

**Comment by JimiC** [comment]
@wooorm Can you point us to any docs that says that `yaml` & `yml` are also valid extensions? From what I could find this will require a [plugin](https://github.com/landakram/remark-parse-yaml) or am I wrong?
